### PR TITLE
Gdpr behavior by integration type

### PR DESCRIPTION
**Pull request by BraslavskiyAndrey**

### CODE REVIEW

File: src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java [hunk 1]
```diff
@@ -143,10 +150,17 @@ public TcfDefinerService(GdprConfig gdprConfig,
...
+        if (requestType == null) {
+            return accountGdprConfig != null && accountGdprConfig.getEnabled() != null
+                    ? accountGdprConfig.getEnabled()
+                    : gdprEnabled;
+        }
+        final EnabledForRequestType enabledForRequestType = accountGdprConfig != null
+                ? accountGdprConfig.getEnabledForRequestType()
+                : null;
+        final Boolean enabled = enabledForRequestType != null ? enabledForRequestType.isEnabledFor(requestType) : null;
+        return enabled != null ? enabled : gdprEnabled;
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java:163]: Probably, it's better to get accountEnabled from retrieved account also.
```
 final Boolean accountGdprEnabled = accountGdprConfig != null && accountGdprConfig.getEnabled() != null;

        if (requestType == null) {
            return ObjectUtils.firstNonNull(accountGdprEnabled, gdprEnabled);
        }

        final EnabledForRequestType enabledForRequestType = accountGdprConfig != null
                ? accountGdprConfig.getEnabledForRequestType()
                : null;
        final Boolean enabledForType = enabledForRequestType != null
                ? enabledForRequestType.isEnabledFor(requestType)
                : null;

        return ObjectUtils.firstNonNull(enabledForType, accountGdprEnabled, gdprEnabled);
```

File: src/test/java/org/prebid/server/privacy/gdpr/TcfDefinerServiceTest.java [hunk 2]
```diff
@@ -112,14 +114,93 @@ public void resultForVendorIdsShouldAllowAllWhenGdprIsDisabled() {
...
+                TcfResponse.of(false, singletonMap("b", PrivacyEnforcementAction.allowAll()), null));
+
+        verifyZeroInteractions(gdprService);
+        verifyZeroInteractions(tcf2Service);
+        verifyZeroInteractions(geoLocationService);
+        verifyZeroInteractions(metrics);
+    }
+
+    @Test
+    public void resultForBidderNamesShouldIgnoreUnknownRequestTypeAndDecideByPbsGdprDisabledProperty() {
```
Comment (DGarbar) [inline src/test/java/org/prebid/server/privacy/gdpr/TcfDefinerServiceTest.java:187]: Don't we need to check account.enabled also ?

Comment (BraslavskiyAndrey) [reply src/test/java/org/prebid/server/privacy/gdpr/TcfDefinerServiceTest.java:187]: No, we don't. Account.enabled only for usersync and setuid. If gdpr for request type is not defined only pbs config checked.

**Comment by BraslavskiyAndrey** [comment]
> Do we need to prepare changes in DB and update tables for this changes ?

Table structure does not require updates. Maybe publisher decide to update their accounts

File: src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java (hunk 1 above)
Comment (BraslavskiyAndrey) [reply src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java:163]: accountGdprConfig.getEnabled() should not have impact on auction typed requests

Comment (BraslavskiyAndrey) [reply src/main/java/org/prebid/server/privacy/gdpr/TcfDefinerService.java]: After discussion with Bret we decide that integration_enabled should override account_enabled. So @DGarbar is right, pull request updated. Thanks

**Pull request merged**